src/main/java/at/ac/tuwien/kr/alpha/grounder/transformation/SumNormalization.java

```diff
@@ -0,0 +1,169 @@
+package at.ac.tuwien.kr.alpha.grounder.transformation;
+
+import at.ac.tuwien.kr.alpha.common.DisjunctiveHead;
+import at.ac.tuwien.kr.alpha.common.Program;
+import at.ac.tuwien.kr.alpha.common.Rule;
+import at.ac.tuwien.kr.alpha.common.atoms.*;
+import at.ac.tuwien.kr.alpha.common.terms.ConstantTerm;
+import at.ac.tuwien.kr.alpha.common.terms.FunctionTerm;
+import at.ac.tuwien.kr.alpha.common.terms.Term;
+import at.ac.tuwien.kr.alpha.common.terms.VariableTerm;
+import at.ac.tuwien.kr.alpha.grounder.Substitution;
+import at.ac.tuwien.kr.alpha.grounder.atoms.EnumerationAtom;
+import at.ac.tuwien.kr.alpha.grounder.parser.InlineDirectives;
+import at.ac.tuwien.kr.alpha.grounder.parser.ProgramParser;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+
+import static at.ac.tuwien.kr.alpha.grounder.transformation.PredicateInternalizer.makePredicatesInternal;
+
+/**
+ * Rewrites #sum aggregates into normal rules.
+ * Note: Currently only in a restricted form.
+ * Copyright (c) 2018, the Alpha Team.
+ */
+public class SumNormalization implements ProgramTransformation {
+
+	private int aggregateCount;
+	private ProgramParser parser = new ProgramParser();
+
+	private Program parse(String program) {
+		return parser.parse(program);
+	}
+
+	@Override
+	public void transform(Program inputProgram) {
+		String summationSubprogram = "interesting_number(R, 1..I1) :- input_number_with_first(R, I, _), I1 = I - 1.\n" +
+			"prefix_subset_sum(R, 0, 0) :- aggregate_function(R, \"#sum\").\n" +
+			"prefix_subset_sum(R, I, S) :- prefix_subset_sum(R, I1, S), I1 = I - 1, interesting_number(R, I).\n" +
+			"prefix_subset_sum(R, I, SF) :- prefix_subset_sum(R, I1, S), I1 = I - 1, SF = S + F, input_number_with_first(R, I, F),  bound(R, K), SF < K.\n" +
+			"output(R, K) :- bound(R, K), K <= 0." +
+			"output(R, K) :- prefix_subset_sum(R, I1, S), I1 = I - 1, input_number_with_first(R, I, F), bound(R, K), K <= S + F.";
+
+		// Connect/Rewrite every aggregate in each rule.
+		ArrayList<Rule> additionalRules = new ArrayList<>();
+		// FIXME: filter facts from additionalRules as they may occur also due to e.g., the bound being a constant and the aggregate occurring as the single body literal.
+		ArrayList<Atom> additionalFacts = new ArrayList<>();
+		int lastAggregateCount = aggregateCount;
+		for (Rule rule : inputProgram.getRules()) {
+			additionalRules.addAll(rewriteAggregates(rule));
+			// Add fact specifying type of aggregate, for example: "aggregate_function(aggregate(1), "#sum")."
+			// FIXME: this only works if there is at most one sum aggregate per original rule.
```

Comment:
Please either resolve remaining inline TODOs before merging to master or create issues to track them.